sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/Distribute.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.appcenter.distribute;
+
+import android.app.Activity;
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import com.microsoft.appcenter.AbstractAppCenterService;
+import com.microsoft.appcenter.channel.Channel;
+import com.microsoft.appcenter.ingestion.models.json.LogFactory;
+import com.microsoft.appcenter.utils.AppCenterLog;
+import com.microsoft.appcenter.utils.async.AppCenterFuture;
+import com.microsoft.appcenter.utils.async.DefaultAppCenterFuture;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.microsoft.appcenter.distribute.DistributeConstants.LOG_TAG;
+import static com.microsoft.appcenter.distribute.DistributeConstants.SERVICE_NAME;
+
+/**
+ * Distribute service.
+ */
+public class Distribute extends AbstractAppCenterService {
+
+    private static Distribute sInstance;
+
+    private static final String DISTRIBUTE_GROUP = "group_distribute";
+
+    /**
+     * Get shared instance.
+     *
+     * @return shared instance.
+     */
+    public static synchronized Distribute getInstance() {
```

Comment:
I remember that in some cases public resume method was used to workaround missing activity if it's started after all events, please add that method too